types/slimerjs/index.d.ts

```diff
@@ -0,0 +1,359 @@
+    userName: string;
+    password: string;
+    XSSAuditingEnabled: boolean;
+    webSecurityEnabled: boolean;
+    resourceTimeout: number;
+}
+
+interface IFileSystem {
+
+    // Properties
+    separator: string;
+    workingDirectory: string;
+
+    // Functions
+
+    // Query Functions
+    list(path: string): string[];
+    absolute(path: string): string;
+    exists(path: string): boolean;
+    isDirectory(path: string): boolean;
+    isFile(path: string): boolean;
+    isAbsolute(path: string): boolean;
+    isExecutable(path: string): boolean;
+    isReadable(path: string): boolean;
+    isWritable(path: string): boolean;
+    isLink(path: string): boolean;
+    readLink(path: string): string;
+
+    // Directory Functions
+    changeWorkingDirectory(path: string): void;
+    makeDirectory(path: string): void;
+    makeTree(path: string): void;
+    removeDirectory(path: string): void;
+    removeTree(path: string): void;
+    copyTree(source: string, destination: string): void;
+
+    // File Functions
+    open(path: string, mode: string): IStream;
+    open(path: string, options: { mode: string; charset?: string; }): IStream;
+    read(path: string): string;
+    write(path: string, content: string, mode: string): void;
+    size(path: string): number;
+    remove(path: string): void;
+    copy(source: string, destination: string): void;
+    move(source: string, destination: string): void;
+    touch(path: string): void;
+}
+
+interface IStream {
+    atEnd(): boolean;
+    close(): void;
+    flush(): void;
+    read(): string;
+    readLine(): string;
+    seek(position: number): void;
+    write(data: string): void;
+    writeLine(data: string): void;
+}
+
+interface IWebServer {
+    port: number;
+    listen(port: number, cb?: (request: IWebServerRequest, response: IWebServerResponse) => void): boolean;
+    listen(ipAddressPort: string, cb?: (request: IWebServerRequest, response: IWebServerResponse) => void): boolean;
+    close(): void;
+}
+
+interface IWebServerRequest {
+    method: string;
+    url: string;
+    httpVersion: number;
+    headers: { [name: string]: string; };
+    post: string;
+    postRaw: string;
+}
+
+interface IWebServerResponse {
+    headers: { [name: string]: string; };
+    setHeader(name: string, value: string): void;
+    header(name: string): string;
+    statusCode: number;
+    setEncoding(encoding: string): void;
+    write(data: string): void;
+    writeHead(statusCode: number, headers?: { [name: string]: string; }): void;
+    close(): void;
+    closeGracefully(): void;
+}
+
+interface ITopLeft {
+    top: number;
+    left: number;
+}
+
+interface ISize {
+    width: number;
+    height: number;
+}
+
+interface IClipRect extends ITopLeft, ISize {
+}
+
+interface ICookie {
+    name: string,
+    value: string,
+    domain?: string,
+    path: string,
+    httponly?: boolean,
+    secure?: boolean,
+    expires?: string,
+    expiry: number
+}
+
+declare module "webpage" {
```

Comment:
The linter must filing errors now. @andy-ms this is intentional, right?

I think `var ... require ...` is fine, because it's already odd that `require` isn't really Node's `require`. The oddness of `var` instead of `import` should call attention to the mismatch.